src/modalities/dataloader/filter_packed_data.py

```diff
@@ -0,0 +1,56 @@
+import pickle
+from pathlib import Path
+from typing import Callable
+
+import numpy as np
+from numpy.typing import NDArray
+from tqdm import tqdm
+
+from modalities.dataloader.create_packed_data import EmbeddedStreamData, update_data_length_in_pre_allocated_header
+from modalities.dataloader.dataset import PackedMemMapDatasetBase
+
+
+def filter_dataset(
+    src_path: Path,
+    dst_path: Path,
+    filter_func: Callable[[tuple[int, dict[str, NDArray[np.int_]]]], bool],
+    sample_key: str = "input_ids",
```

Comment:
It looks to me  as if we could hardcode this lcoally, as this is only used internally.